pkg/test/test.go

```diff
@@ -0,0 +1,38 @@
+// Package test implements the common structs used for defining
+// testcases.
+package test
+
+// Config struct is used to pass arbitrary information into the tests
+// such as IP address, expected FRU content, expected sensors
+type Config struct {
+
+	Host string;
+	Username string;
+	Password string;
+
+	// ... to be extended
+}
+
+// Result struct is used for returning test results from running a test
+type Result struct {
+
+	// what was tested
+	Name string
+
+	// != nil in case something went wrong,
+	// this should contain a descriptive string for the logs.
+	// e.g. "while running command X, expected A but output was B'
+	Error error
```

Comment:
We also need some `output string` entry here so that we can store `stdout` or `stderr`.

Reply:
This is complicated since not every test result can be mapped to a command execution. But i can add the string so we have it just in case.